src/ray/raylet/agent_manager.cc

```diff
@@ -226,29 +203,25 @@ void AgentManager::CreateRuntimeEnv(
           }
 
         } else {
-          // TODO(sang): Invoke a callback if it fails more than X times.
           RAY_LOG(INFO)
               << "Failed to create the runtime env: " << serialized_runtime_env
               << ", status = " << status
-              << ", maybe there are some network problems, will retry it later.";
-          delay_executor_(
-              [this,
-               job_id,
-               serialized_runtime_env,
-               serialized_allocated_resource_instances,
-               callback = std::move(callback)] {
-                CreateRuntimeEnv(job_id,
-                                 serialized_runtime_env,
-                                 serialized_allocated_resource_instances,
-                                 callback);
-              },
-              RayConfig::instance().agent_manager_retry_interval_ms());
+              << ", maybe there are some network problems, will fail the request.";
+          callback(false, "", "Failed to request agent.");
         }
       });
 }
 
 void AgentManager::DeleteURIs(const std::vector<std::string> &uris,
                               DeleteURIsCallback callback) {
+  if (disable_agent_client_) {
+    RAY_LOG(ERROR)
+        << "Failed to delete runtime env URIs because the agent client is disabled. You "
+           "see this error message maybe because the grpc port of agent came into "
+           "conflict. Please see `dashboard_agent.log` to get more details.";
```

Comment:
```suggestion
        << "Failed to delete runtime env URIs because the agent client is disabled. You "
           "might see this error message because the gRPC port of the agent came into "
           "conflict. Please see `dashboard_agent.log` to get more details.";
```